OSDOCS-4444 and OSDOCS-5270: adding Running Apps book and operators content (#55640)
openshift/openshift-docs: microshift_running_apps/microshift-operators.adoc

:_content-type: ASSEMBLY
[id="operators-with-microshift"]
= Using Operators
include::_attributes/attributes-microshift.adoc[]
:context: operators-microshift

toc::[]

> Operators integrate with Kubernetes APIs and CLI tools such as `kubectl` and `oc` commands. They provide the means to manage applications and the resources used by applications, such as deploying a database or message bus.

Reviewer: ```suggestion
Operators integrate with Kubernetes APIs and CLI tools such as `kubectl` and `oc`. They can manage applications and their resources, such as deploying a database or message bus.
```
Minimalism suggestion

> Operators integrate with Kubernetes APIs and CLI tools such as `kubectl` and `oc` commands. They provide the means of monitoring applications, performing health checks, managing over-the-air (OTA) updates, and ensuring that applications remain in your specified state.

Reviewer 2: These examples are all relevant for OCP, but not so much for MicroShift. Maybe we can come up with some other examples. I expect the sorts of operators people will deploy to MicroShift will be more focused on managing applications or resources used by applications (deploying a database or message bus, for example).

Author: We can just start with your examples and iterate?

> Operators integrate with Kubernetes APIs and CLI tools such as `kubectl` and `oc`. Operators provide the means to manage applications and their resources, such as deploying a database or message bus. The Operator Lifecycle Manager (OLM) is not supported by {product-title}, but you can install Operators with manifests.

Reviewer 2: ```suggestion
Operators integrate with Kubernetes APIs and CLI tools such as `kubectl` and `oc`. Operators provide the means to manage applications and their resources, such as deploying a database or message bus. The Operator Lifecycle Manager (OLM) is not included with {product-title}, but you can install Operators with manifests.
```

OLM works on MicroShift, we just don't include it or expect users to want to use it.

If you want to get into the reasoning and focus more on the "what works" instead of what does not work, we could say something like "In order to minimize the footprint of {product-title}, Operators are installed directly with manifests instead of through the Operator Lifecycle Manager (OLM)."

> Operators offer a more granular configuration experience. You configure each component by modifying the API that the Operator exposes instead of modifying a global configuration file. Even though {product-title} applications are often deployed in static environments, you can install Operators with manifests for your use.

Reviewer: ```suggestion
Operators offer a more granular configuration experience. You configure each component by modifying the API that the Operator exposes instead of modifying a global configuration file. Even though {product-title} applications are often deployed in static environments, you can install Operators with manifests.
```
Minimalism suggestion. It might be worth explaining how manifests are different than static environments - it isn't clear to me what the distinction is.

Author: I expanded this a bit. So what's happening here are two things: 1) static environments don't really need operators -- 'set and forget' is the general direction for applications running on MicroShift. And two, in OCP, Operators are installed and used through OLM (the Operator Lifecycle Manager). Since this is a very lightweight Kubernetes, using OLM isn't an option. You can install the Operators you want with manifests, though, if you need them.
